### PR TITLE
NT-2037: Pagination Error Cell displayed at the same time as the comments cells 

### PR DESCRIPTION
**Pull request by Arkariang**
# 📲 What

- Subtask to add the new cell to handle the pagination error state
- ⚠️  the actual UI for the cell will come on next PR's this is just displaying two types of cell within the same RecyclerView and tidying up together the pagination errors logic.

# 🤔 Why
- Error when pagination designs required a a new cell within the actual recyclerview

# 👀 See

errorCellScreenShoot

https://user-images.githubusercontent.com/4083656/121607613-32bfa600-ca05-11eb-8ace-7e1ba5b75586.mp4

|  |  |

# 📋 QA
- Load comments for a project with many comments, turn off you connection, scroll to the bottom and try to paginate, you'll see the placeholder error cell appearing.
- recover connectivity, try to scroll again, the cell will disappear and you will see more comments added to the list.

# Story 📖

[NT-2037](https://kickstarter.atlassian.net/browse/NT-2037)

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/adapters/CommentsAdapter.kt [hunk 1]
```diff
@@ -4,27 +4,52 @@ import android.view.LayoutInflater
...
+import com.kickstarter.ui.viewholders.EmptyViewHolder
 import com.kickstarter.ui.viewholders.KSViewHolder
 
 class CommentsAdapter(private val delegate: Delegate) : KSListAdapter() {
     interface Delegate : EmptyCommentsViewHolder.Delegate, CommentCardViewHolder.Delegate
 
-    @LayoutRes
-    override fun layout(sectionRow: SectionRow): Int {
-        return R.layout.item_comment_card
+    init {
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/adapters/CommentsAdapter.kt:19]: @sunday-okpoluaefe this is gonna generate conflicts with your current work in progress, let me know if any doubt we can pair program for solving it

File: app/src/main/res/layout/activity_comments_layout.xml [hunk 2]
```diff
@@ -72,13 +73,13 @@
                 android:background="@color/kds_white"
                 app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
                 tools:listitem="@layout/comment_card" />
-
         </androidx.swiperefreshlayout.widget.SwipeRefreshLayout>
 
         <ProgressBar
```
Comment (Arkariang) [inline app/src/main/res/layout/activity_comments_layout.xml:78]: @hadia we could use the same new cell I just created for the loading indicator, think about it!

**Review by leighdouglas — APPROVED** [review]
lgtm! 🚀

File: app/src/main/res/layout/activity_comments_layout.xml (hunk 2 above)
Comment (hadia) [reply app/src/main/res/layout/activity_comments_layout.xml:78]: yes agree

**Pull request merged**